docs/internal/contributing/how-to-convert-config-to-per-tenant-limit.md

```diff
@@ -0,0 +1,181 @@
+# Converting a Mimir Config Option to a Per-Tenant Limit
+
+When running Mimir with multi-tenancy enabled, usage patterns among tenants may diverge to the point that an option needs to be configured independently for different tenants.
+
+The values set in the `limits` section of the [Mimir config file](https://grafana.com/docs/mimir/latest/references/configuration-parameters/#limits) serve as global config for all tenants.
+These global limits can then be overridden for specified tenants via the `overrides` section of the [runtime configuration](https://grafana.com/docs/mimir/latest/configure/about-runtime-configuration/#runtime-configuration-of-per-tenant-limits).
+
+If a config option is not present in the `limits` section, it cannot be overridden on a per-tenant basis.
+If the need arises to override an existing config option for individual tenants, we must follow a deprecation process to move the option into the `limits` section.
+
+## Deprecation and Migration Process
+
+We need to ensure that:
+
+- the existing config option is deprecated
+- the new limits config option is added
+- the new limits config option can be set and overridden in a backwards-compatible manner until deprecation is complete
+- all docs are updated
+
+We will use the [Mimir PR #5312](https://github.com/grafana/mimir/pull/5312) as a reference example, in which we moved the query-frontend option `cache_unaligned_requests` from the `frontend` config section to `limits`.
+[PR #4287](https://github.com/grafana/mimir/pull/4287) is also available for reference.
+
+## Deprecating the Existing Config Option
+
+1. Move the config option description JSON object from its previous location to the "limits" block of `cmd/mimir/config-descriptor.json`.
```

Comment:
This file is automatically generated when `make reference-help` is run. It shouldn't be modified manually.